packages/ui/src/composites/wui-chip-button/index.ts

```diff
@@ -0,0 +1,46 @@
+import { html, LitElement } from 'lit'
+import { property } from 'lit/decorators.js'
+import '../../components/wui-icon/index.js'
+import '../../components/wui-image/index.js'
+import '../../components/wui-text/index.js'
+import { elementStyles, resetStyles } from '../../utils/ThemeUtil.js'
+import type { ChipType, IconType } from '../../utils/TypeUtil.js'
+import { customElement } from '../../utils/WebComponentsUtil.js'
+import styles from './styles.js'
+
+@customElement('wui-chip-button')
+export class WuiChipButton extends LitElement {
+  public static override styles = [resetStyles, elementStyles, styles]
+
+  // -- State & Properties -------------------------------- //
+  @property() public variant: ChipType = 'fill'
+
+  @property() public imageSrc = ''
+
+  @property({ type: Boolean }) public disabled = false
+
+  @property() public icon: IconType = 'externalLink'
+
+  @property() public text = ''
+
+  // -- Render -------------------------------------------- //
+  public override render() {
+    const isSmall =
+      this.variant === 'success' || this.variant === 'transparent' || this.variant === 'shadeSmall'
```

Comment:
just wondering if it would be better to have a `small` prop or even having a `textVariant` prop directly